service-mirror.go

```diff
@@ -2,23 +2,31 @@ package main
 
 import (
 	"context"
+	ctx "context"
 	"flag"
-	"fmt"
 	"os"
 	"path/filepath"
-	"strings"
 
 	"github.com/sirupsen/logrus"
-	corev1 "k8s.io/api/core/v1"
-	apiError "k8s.io/apimachinery/pkg/api/errors"
-	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
-	"k8s.io/apimachinery/pkg/labels"
-	"k8s.io/apimachinery/pkg/selection"
 	"k8s.io/client-go/kubernetes"
-	"k8s.io/client-go/tools/clientcmd"
+	"k8s.io/client-go/tools/cache"
+	client "k8s.io/client-go/tools/clientcmd"
 	"k8s.io/client-go/util/homedir"
 )
 
+// Make sure all global services lies in only one namespace.
+const GLOBAL_SVC_NAMESPACE = "default"
+
+type Context struct {
+	ctx.Context
+	kubernetes.Clientset
```

Comment:
I would do a field name on these and then embed in the watcher so we can do watcher.client or watcher.ctx

Reply:
This is still part of context struct, it had named previously so it will be again like svc.ctx.client or svc.ctx.ctx.

Reply:
If you embed you can refer to the field directly. If your watcher holds an embdded context you can do watcher.client